client/src/main/proto/spine/base/field_path.proto

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright 2016, TeamDev Ltd. All rights reserved.
+//
+// Redistribution and use in source and/or binary forms, with or without
+// modification, must retain the above copyright notice and the following
+// disclaimer.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+syntax = "proto3";
+
+package spine.base;
+
+option java_package = "org.spine3.base";
+option java_outer_classname = "FieldPathProto";
+option java_multiple_files = true;
+option java_generate_equals_and_hash = true;
+
+
+// A field path, which consists of simple message field names, e.g. ["field1", "field2", ...].
```

Comment:
Have a look at description of `FieldMask` for ideas on describing the field patch concept. Give an example of nested data structure, which may contain a fictitious data structures `CustomerAccount`, which contains `user` record of type `User`, which in turn has `first_name` field of type `string`. In this case, the path would be `user.first_name`. Being structurally represented in `FieldPath` it would have two entries.

Just having `field1` and `field2` in the example makes it hard to understand.